data_hub_api/docmaps/v2/sql/docmaps_index.sql

```diff
@@ -302,11 +302,42 @@ t_rp_meca_path_update AS (
   WHERE rn=1
 ),
 
+t_manual_preprint_match_for_published_date AS (
+  SELECT
+    * EXCEPT(rn),
+  FROM
+  (
+    SELECT 
+      *,
+      ROW_NUMBER() OVER(PARTITION BY long_manuscript_identifier ORDER BY imported_timestamp DESC) AS rn
+    FROM `elife-data-pipeline.prod.unmatched_manuscripts`
+    WHERE preprint_published_at_date IS NOT NULL 
+      AND preprint_published_at_date != ''
+  )
+  WHERE rn=1
+),
+
+t_europepmc_preprint_publication_date AS (
+  SELECT
+    response.doi,
+    response.firstPublicationDate,
+    url_list.url
+  FROM `elife-data-pipeline.prod.v_latest_europepmc_preprint_servers_response`  AS response
+  LEFT JOIN UNNEST(fullTextUrlList.fullTextUrl) AS url_list
+  -- This filter prevents date inaccuracies due to missing version numbers in OSF and arXiv DOIs
+  WHERE LOWER(url_list.url) NOT LIKE '%/osf.io/%'
```

Comment:
can't we see that from the DOI?

Reply:
DOI or URL does not contain version info for OSF and arXiv preprints. If that is you are asking?